src/components/Modal.js

```diff
@@ -80,6 +78,11 @@ class Modal extends Component {
 
         if (this.props.isOpen && !this.props.isStatic) {
             if (this.props.onCancel) {
+
+                // Ensure that onToggle gets called.
+                this.onToggle(false, this.props);
+
+                // Call the user supplied function.
```

Comment:
Isn't onToggle called anyway, only after the next render though?

E.g.

1. User presses Escape
2. modal.onCancel gets called which triggers the `props.onCancel` hook
3. userland reacts to `props.onCancel` hook and sets modal open state to false
4. modal receives a props change where `isOpen` is now false
5. onToggle is called via `componentWillReceiveProps` 

Calling onToggle directly here removes the control from the userland, e.g. it would not be possible to `"ignore"` some onCancel requests anymore.